src/os/open.zig

```diff
@@ -77,3 +75,125 @@ const OpenCommand = struct {
     child: std.process.Child,
     wait: bool = false,
 };
+
+/// Use `xdg-open` to open a URL using the default application.
+///
+/// Any output on stderr is logged as a warning in the application logs. Output
+/// on stdout is ignored.
+pub fn openUrlLinux(
```

Comment:
I don't really love the name here... Maybe `openUrlXdg` makes more sense - this would also allow us to add something like `openUrlPortal`. Or maybe, we should just add a "method" enum as the parameter to `open`, that looks something like

```zig
pub const OpenMethod = enum {
    /// Use macOS's builtin `open` utility to open URLs.
    open,
    /// Use the `xdg-open` utility to open URLs. Available on most Linux distributions.
    xdg_open,
    /// Use the OpenURL method of the XDG Desktop Portal on Linux.
    xdg_desktop_portal,
}
```